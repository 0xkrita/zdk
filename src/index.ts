import { GraphQLClient } from 'graphql-request';
import {
  getSdk,
  Chain,
  Network,
  NetworkInput,
  SortDirection,
  TokenInput,
  TokensQueryInput,
  TokensQueryFilter,
  CollectionsQueryInput,
  CollectionSortKeySortInput,
  TokenSortInput,
  CollectionSortKey,
  TokenSortKey,
  AggregateAttributesQueryVariables,
  OwnersByCountQueryVariables,
  SalesVolumeQueryVariables,
  EventsQueryInput,
  EventsQueryFilter,
  EventSortKeySortInput,
  EventSortKey,
  MarketSortKeySortInput,
  MarketsQueryFilter,
  MarketsQueryInput,
  MarketSortKey,
  MintsQueryFilter,
  MintsQueryInput,
  MintSortKey,
  MintSortKeySortInput,
  FloorPriceQueryVariables,
  NftCountQueryVariables,
  FullTextSearchQueryVariables,
  TokenQueryVariables,
  SaleSortKey,
  SalesQueryInput,
  SaleSortKeySortInput,
  SalesQueryFilter,
} from './queries/queries-sdk';

// Export chain and network for API users
export { Chain as ZDKChain, Network as ZDKNetwork };

export type OverrideNetworksOption = NetworkInput[];

export type OverridePaginationOptions = {
  limit?: number;
  offset?: number;
};

type SharedQueryParams = {
  networks?: NetworkInput[];
  pagination?: OverridePaginationOptions;
};

type CollectionsQueryArgs = {
  where: CollectionsQueryInput;
  includeFullDetails?: boolean;
  sort: CollectionSortKeySortInput;
} & SharedQueryParams;

type TokensQueryArgs = {
  where: TokensQueryInput;
  filter?: TokensQueryFilter;
  sort?: TokenSortInput;
  includeFullDetails?: boolean;
  includeSalesHistory?: boolean;
} & SharedQueryParams;

type EventsQueryArgs = {
  where: EventsQueryInput;
  sort?: EventSortKeySortInput;
  filter?: EventsQueryFilter;
} & SharedQueryParams;

type MarketQueryArgs = {
  where: MarketsQueryInput;
  sort: MarketSortKeySortInput;
  filter: MarketsQueryFilter;
  includeFullDetails: boolean;
} & SharedQueryParams;

type MintsQueryArgs = {
  where: MintsQueryInput;
  sort: MintSortKeySortInput;
  filter: MintsQueryFilter;
  includeFullDetails: boolean;
} & SharedQueryParams;

type SalesQueryArgs = {
  where: SalesQueryInput;
  sort: SaleSortKeySortInput;
  filter: SalesQueryFilter;
  includeFullDetails: boolean;
} & SharedQueryParams;

export interface ListOptions<SortInput> {
  networks?: OverrideNetworksOption;
  pagination?: OverridePaginationOptions;
  sort?: SortInput;
  includeFullDetails?: boolean;
}

export interface AggregateOptions {
  networks?: OverrideNetworksOption;
}

export type TokenQueryList = TokenInput;

type OptionalNetwork<K> = Omit<K, 'networks'> & {
  networks?: OverrideNetworksOption;
};

const DEFAULT_PROD_ENDPOINT = 'https://api.zora.co/graphql';

export class ZDK {
  endpoint: string;
  defaultNetworks: OverrideNetworksOption;
  defaultMaxPageSize: number = 200;

  public sdk: ReturnType<typeof getSdk>;

  constructor(
    endpoint: string = DEFAULT_PROD_ENDPOINT,
    networks: OverrideNetworksOption = [
      { network: Network.Ethereum, chain: Chain.Mainnet },
    ]
  ) {
    this.endpoint = endpoint;
    this.defaultNetworks = networks;
    this.sdk = getSdk(new GraphQLClient(this.endpoint));
  }

  private getNetworksOption = (networks?: OverrideNetworksOption) => {
    return {
      networks: networks ?? this.defaultNetworks,
    };
  };

  private getPaginationOptions = ({ limit, offset }: OverridePaginationOptions = {}) => {
    return {
      pagination: {
        limit: limit || this.defaultMaxPageSize,
        offset: offset || 0,
      },
    };
  };

  public tokens = async ({
    where,
    filter,
    pagination,
    networks,
    sort,
    includeFullDetails = false,
    includeSalesHistory = false,
  }: TokensQueryArgs) =>
    this.sdk.tokens({
      where,
      filter,
      sort: {
        sortDirection: sort?.sortDirection || SortDirection.Asc,
        sortKey: sort?.sortKey || TokenSortKey.Transferred,
        sortAxis: sort?.sortAxis,
      },
      includeFullDetails,
      includeSalesHistory,
      ...this.getPaginationOptions(pagination),
      ...this.getNetworksOption(networks),
    });

  public token = async ({
    token,
    network = this.defaultNetworks[0],
    includeFullDetails,
  }: OptionalNetwork<TokenQueryVariables>) =>
    await this.sdk.token({
      token,
      network,
      includeFullDetails,
    });

  public events = async ({
    networks,
    filter,
    pagination,
    sort,
    where,
  }: EventsQueryArgs) =>
    this.sdk.events({
      filter,
      where,
      ...this.getPaginationOptions(pagination),
      ...this.getNetworksOption(networks),
      sort: {
        sortDirection: sort?.sortDirection || SortDirection.Desc,
        sortKey: sort?.sortKey || EventSortKey.Created,
      },
    });

  public markets = async ({
    networks,
    filter,
    pagination,
    sort,
    where,
    includeFullDetails = false,
  }: MarketQueryArgs) =>
    this.sdk.markets({
      filter,
      where,
      includeFullDetails,
      ...this.getPaginationOptions(pagination),
      ...this.getNetworksOption(networks),
      sort: {
        sortDirection: sort?.sortDirection || SortDirection.Desc,
        sortKey: sort?.sortKey || MarketSortKey.None,
      },
    });

  public mints = async ({
    networks,
    filter,
    pagination,
    sort,
    where,
    includeFullDetails = false,
  }: MintsQueryArgs) =>
    this.sdk.mints({
      filter,
      where,
      ...this.getPaginationOptions(pagination),
      ...this.getNetworksOption(networks),
      sort: {
        sortDirection: sort?.sortDirection || SortDirection.Desc,
        sortKey: sort?.sortKey || MintSortKey.None,
      },
      includeFullDetails,
    });

  public sales = async ({
    where,
    pagination,
    networks,
    sort,
    filter,
    includeFullDetails = false,
  }: SalesQueryArgs) =>
    this.sdk.sales({
      where,
      filter,
      ...this.getPaginationOptions(pagination),
      ...this.getNetworksOption(networks),
      sort: {
        sortDirection: sort?.sortDirection || SortDirection.Desc,
        sortKey: sort?.sortKey || SaleSortKey.Time,
      },
      includeFullDetails,
    });

  public collections = async ({
    where,
    pagination,
    networks,
    sort,
    includeFullDetails = false,
  }: CollectionsQueryArgs) =>
    this.sdk.collections({
      where,
      includeFullDetails,
      ...this.getPaginationOptions(pagination),
      ...this.getNetworksOption(networks),
      sort: {
        sortDirection: sort?.sortDirection || SortDirection.Desc,
        sortKey: sort?.sortKey || CollectionSortKey.Created,
      },
    });

  public ownersByCount = async ({
    where,
    pagination,
    networks = this.defaultNetworks,
  }: OptionalNetwork<OwnersByCountQueryVariables>) =>
    this.sdk.ownersByCount({
      where,
      networks,
      pagination,
    });

  public aggregateAttributes = async ({
    where,
    networks = this.defaultNetworks,
  }: OptionalNetwork<AggregateAttributesQueryVariables>) =>
    this.sdk.aggregateAttributes({ where, networks });

  public salesVolume = async ({
    where,
    networks = this.defaultNetworks,
    timeFilter,
  }: OptionalNetwork<SalesVolumeQueryVariables>) =>
    this.sdk.salesVolume({
      where,
      networks,
      timeFilter,
    });

  public ownerCount = async ({
    where,
    networks = this.defaultNetworks,
  }: OptionalNetwork<OwnersByCountQueryVariables>) =>
    this.sdk.ownerCount({
      where,
      networks,
    });

  public floorPrice = async ({
    where,
    networks = this.defaultNetworks,
  }: OptionalNetwork<FloorPriceQueryVariables>) =>
    this.sdk.floorPrice({
      where,
      networks,
    });

  public nftCount = async ({
    where,
    networks = this.defaultNetworks,
  }: OptionalNetwork<NftCountQueryVariables>) =>
    this.sdk.nftCount({
      where,
      networks,
    });

  public search = async ({ pagination, query, filter }: FullTextSearchQueryVariables) =>
    this.sdk.fullTextSearch({
      pagination,
      filter,
      query,
    });
}
