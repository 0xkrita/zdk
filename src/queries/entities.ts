export {TokenContentMedia} from './queries-sdk';