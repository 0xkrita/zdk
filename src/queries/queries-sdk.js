'use strict';
var __makeTemplateObject =
  (this && this.__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', { value: raw });
    } else {
      cooked.raw = raw;
    }
    return cooked;
  };
var __assign =
  (this && this.__assign) ||
  function () {
    __assign =
      Object.assign ||
      function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
          s = arguments[i];
          for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
      };
    return __assign.apply(this, arguments);
  };
exports.__esModule = true;
exports.getSdk =
  exports.GetTokenDocument =
  exports.GetCollectionsDocument =
  exports.GetTokenContractDocument =
  exports.Network =
    void 0;
var graphql_tag_1 = require('graphql-tag');
var Network;
(function (Network) {
  Network['Ethereum'] = 'ETHEREUM';
  Network['Flow'] = 'FLOW';
  Network['Solana'] = 'SOLANA';
})((Network = exports.Network || (exports.Network = {})));
exports.GetTokenContractDocument = (0, graphql_tag_1['default'])(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      [
        '\n    query getTokenContract($addresses: [String!]) {\n  tokenContracts(addresses: $addresses) {\n    name\n    description\n  }\n}\n    ',
      ],
      [
        '\n    query getTokenContract($addresses: [String!]) {\n  tokenContracts(addresses: $addresses) {\n    name\n    description\n  }\n}\n    ',
      ]
    ))
);
exports.GetCollectionsDocument = (0, graphql_tag_1['default'])(
  templateObject_2 ||
    (templateObject_2 = __makeTemplateObject(
      [
        '\n    query getCollections($addresses: [String!]) {\n  collections(addresses: $addresses) {\n    name\n    description\n  }\n}\n    ',
      ],
      [
        '\n    query getCollections($addresses: [String!]) {\n  collections(addresses: $addresses) {\n    name\n    description\n  }\n}\n    ',
      ]
    ))
);
exports.GetTokenDocument = (0, graphql_tag_1['default'])(
  templateObject_3 ||
    (templateObject_3 = __makeTemplateObject(
      [
        '\n    query getToken($addresses: [String!]) {\n  tokens(addresses: $addresses) {\n    name\n    metadata\n  }\n}\n    ',
      ],
      [
        '\n    query getToken($addresses: [String!]) {\n  tokens(addresses: $addresses) {\n    name\n    metadata\n  }\n}\n    ',
      ]
    ))
);
var defaultWrapper = function (action, _operationName) {
  return action();
};
function getSdk(client, withWrapper) {
  if (withWrapper === void 0) {
    withWrapper = defaultWrapper;
  }
  return {
    getTokenContract: function (variables, requestHeaders) {
      return withWrapper(function (wrappedRequestHeaders) {
        return client.request(
          exports.GetTokenContractDocument,
          variables,
          __assign(__assign({}, requestHeaders), wrappedRequestHeaders)
        );
      }, 'getTokenContract');
    },
    getCollections: function (variables, requestHeaders) {
      return withWrapper(function (wrappedRequestHeaders) {
        return client.request(
          exports.GetCollectionsDocument,
          variables,
          __assign(__assign({}, requestHeaders), wrappedRequestHeaders)
        );
      }, 'getCollections');
    },
    getToken: function (variables, requestHeaders) {
      return withWrapper(function (wrappedRequestHeaders) {
        return client.request(
          exports.GetTokenDocument,
          variables,
          __assign(__assign({}, requestHeaders), wrappedRequestHeaders)
        );
      }, 'getToken');
    },
  };
}
exports.getSdk = getSdk;
var templateObject_1, templateObject_2, templateObject_3;
