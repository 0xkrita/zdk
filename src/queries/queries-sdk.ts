import { GraphQLClient } from 'graphql-request';
import * as Dom from 'graphql-request/dist/types.dom';
import gql from 'graphql-tag';
export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & {
  [SubKey in K]?: Maybe<T[SubKey]>;
};
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & {
  [SubKey in K]: Maybe<T[SubKey]>;
};
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  /** The GenericScalar scalar type represents a generic GraphQL scalar value that could be: List or Object. */
  JSONScalar: any;
};

export type Collection = {
  __typename?: 'Collection';
  address: Scalars['String'];
  name: Scalars['String'];
  description: Scalars['String'];
  totalSupply?: Maybe<Scalars['Int']>;
};

export enum Network {
  Ethereum = 'ETHEREUM',
  Flow = 'FLOW',
  Solana = 'SOLANA',
}

export type RootQuery = {
  __typename?: 'RootQuery';
  /** Get all collections */
  collections: Array<Collection>;
  /** Get all token contracts */
  tokenContracts: Array<TokenContract>;
  /** Get all tokens */
  tokens: Array<Token>;
};

export type RootQueryCollectionsArgs = {
  addresses?: Maybe<Array<Scalars['String']>>;
};

export type RootQueryTokenContractsArgs = {
  addresses?: Maybe<Array<Scalars['String']>>;
};

export type RootQueryTokensArgs = {
  addresses?: Maybe<Array<Scalars['String']>>;
};

export type Token = {
  __typename?: 'Token';
  tokenAddress: Scalars['String'];
  tokenId: Scalars['String'];
  network: Network;
  tokenUrl: Scalars['String'];
  tokenUrlMimeType?: Maybe<Scalars['String']>;
  owner: Scalars['String'];
  tokenContract: TokenContract;
  mintBlockNumber: Scalars['Int'];
  name?: Maybe<Scalars['String']>;
  description?: Maybe<Scalars['String']>;
  externalURL?: Maybe<Scalars['String']>;
  metadata?: Maybe<Scalars['JSONScalar']>;
  attributes?: Maybe<Array<TokenAttribute>>;
  minter?: Maybe<Scalars['String']>;
  lastRefreshTime?: Maybe<Scalars['Int']>;
};

export type TokenAttribute = {
  __typename?: 'TokenAttribute';
  traitType: Scalars['String'];
  name?: Maybe<Scalars['String']>;
  value?: Maybe<Scalars['String']>;
  role?: Maybe<Scalars['String']>;
  displayType?: Maybe<Scalars['String']>;
  maxValue?: Maybe<Scalars['String']>;
  traitCount?: Maybe<Scalars['Int']>;
  shares?: Maybe<Scalars['Int']>;
};

export type TokenContract = {
  __typename?: 'TokenContract';
  tokenAddress: Scalars['String'];
  network: Scalars['String'];
  chain: Scalars['Int'];
  name: Scalars['String'];
  symbol: Scalars['String'];
  totalSupply?: Maybe<Scalars['Int']>;
  schema: Scalars['String'];
  description: Scalars['String'];
  creator: Scalars['String'];
  iconURL: Scalars['String'];
  externalURL: Scalars['String'];
};

export const GetTokenContractDocument = gql`
  query getTokenContract($addresses: [String!]) {
    tokenContracts(addresses: $addresses) {
      name
      description
    }
  }
`;
export const GetCollectionsDocument = gql`
  query getCollections($addresses: [String!]) {
    collections(addresses: $addresses) {
      name
      description
    }
  }
`;
export const GetTokenDocument = gql`
  query getToken($addresses: [String!]) {
    tokens(addresses: $addresses) {
      name
      metadata
    }
  }
`;

export type SdkFunctionWrapper = <T>(
  action: (requestHeaders?: Record<string, string>) => Promise<T>,
  operationName: string
) => Promise<T>;

const defaultWrapper: SdkFunctionWrapper = (action, _operationName) => action();

export function getSdk(
  client: GraphQLClient,
  withWrapper: SdkFunctionWrapper = defaultWrapper
) {
  return {
    getTokenContract(
      variables?: GetTokenContractQueryVariables,
      requestHeaders?: Dom.RequestInit['headers']
    ): Promise<GetTokenContractQuery> {
      return withWrapper(
        (wrappedRequestHeaders) =>
          client.request<GetTokenContractQuery>(GetTokenContractDocument, variables, {
            ...requestHeaders,
            ...wrappedRequestHeaders,
          }),
        'getTokenContract'
      );
    },
    getCollections(
      variables?: GetCollectionsQueryVariables,
      requestHeaders?: Dom.RequestInit['headers']
    ): Promise<GetCollectionsQuery> {
      return withWrapper(
        (wrappedRequestHeaders) =>
          client.request<GetCollectionsQuery>(GetCollectionsDocument, variables, {
            ...requestHeaders,
            ...wrappedRequestHeaders,
          }),
        'getCollections'
      );
    },
    getToken(
      variables?: GetTokenQueryVariables,
      requestHeaders?: Dom.RequestInit['headers']
    ): Promise<GetTokenQuery> {
      return withWrapper(
        (wrappedRequestHeaders) =>
          client.request<GetTokenQuery>(GetTokenDocument, variables, {
            ...requestHeaders,
            ...wrappedRequestHeaders,
          }),
        'getToken'
      );
    },
  };
}
export type Sdk = ReturnType<typeof getSdk>;
export type GetTokenContractQueryVariables = Exact<{
  addresses?: Maybe<Array<Scalars['String']> | Scalars['String']>;
}>;

export type GetTokenContractQuery = { __typename?: 'RootQuery' } & {
  tokenContracts: Array<
    { __typename?: 'TokenContract' } & Pick<TokenContract, 'name' | 'description'>
  >;
};

export type GetCollectionsQueryVariables = Exact<{
  addresses?: Maybe<Array<Scalars['String']> | Scalars['String']>;
}>;

export type GetCollectionsQuery = { __typename?: 'RootQuery' } & {
  collections: Array<
    { __typename?: 'Collection' } & Pick<Collection, 'name' | 'description'>
  >;
};

export type GetTokenQueryVariables = Exact<{
  addresses?: Maybe<Array<Scalars['String']> | Scalars['String']>;
}>;

export type GetTokenQuery = { __typename?: 'RootQuery' } & {
  tokens: Array<{ __typename?: 'Token' } & Pick<Token, 'name' | 'metadata'>>;
};
