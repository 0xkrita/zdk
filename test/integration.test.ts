import { Chain, Network } from '../src/queries/queries-sdk';
import { ZDK } from '../src/zdk';

describe('zdk', () => {
  let zdk: ZDK;
  beforeEach(() => {
    zdk = new ZDK(process.env.ZDK_ENDPOINT!, Network.Ethereum, Chain.Mainnet);
  });
  it('should fetch localhost collections empty object', async () => {
    expect(
      await zdk.tokens(['0xCa21d4228cDCc68D4e23807E5e370C07577Dd152'], {
        pagination: { limit: 2 },
      })
    ).toMatchSnapshot();
  });
  it('should fetch localhost tokens empty object', async () => {
    expect(
      await zdk.token('0xCa21d4228cDCc68D4e23807E5e370C07577Dd152', '12')
    ).toMatchSnapshot();
  });
  it('should fetch localhost token contract empty object', async () => {
    expect(
      await zdk.tokensSummary(['0xCa21d4228cDCc68D4e23807E5e370C07577Dd152'], {pagination: {limit: 4}})
    ).toMatchSnapshot();
  });
});
