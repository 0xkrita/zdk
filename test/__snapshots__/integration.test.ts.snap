// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zdk queries long token information 1`] = `
Object {
  "collections": Object {
    "nodes": Array [
      Object {
        "address": "0x5180db8f5c931aae63c74266b211f580155ecac8",
        "description": "",
        "name": "Crypto Coven",
        "symbol": "WITCH",
        "totalSupply": null,
      },
    ],
    "pageInfo": Object {
      "limit": 200,
      "offset": 0,
    },
    "totalCount": 1,
  },
}
`;

exports[`zdk queries sort collection information 1`] = `
Object {
  "collections": Object {
    "nodes": Array [
      Object {
        "address": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
        "description": "",
        "name": "Zorbs",
        "symbol": "ZORB",
        "totalSupply": 56741,
      },
    ],
    "pageInfo": Object {
      "limit": 200,
      "offset": 0,
    },
    "totalCount": 1,
  },
}
`;

exports[`zdk should fetch localhost collections empty object 1`] = `
Object {
  "markets": Array [
    Object {
      "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
      "marketAddress": "0xce6cef2a9028e1c3b21647ae3b4251038109f42a",
      "marketType": "V3_ASK",
      "networkInfo": Object {
        "chain": "MAINNET",
        "network": "ETHEREUM",
      },
      "price": Object {
        "blockNumber": 14036921,
        "ethPrice": Object {
          "decimal": 0.05,
          "raw": "50000000000000000",
        },
        "nativePrice": Object {
          "currency": Object {
            "address": "0x0000000000000000000000000000000000000000",
            "decimals": 18,
            "name": "ETH",
          },
          "decimal": 0.05,
        },
        "usdcPrice": Object {
          "decimal": 156.40084904014552,
          "raw": "156400849040145514496",
        },
      },
      "status": "ACTIVE",
    },
  ],
  "sales": Object {},
  "token": Object {
    "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
    "content": null,
    "description": null,
    "image": null,
    "lastRefreshTime": null,
    "mintInfo": Object {
      "mintContext": Object {
        "blockNumber": 13927877,
        "blockTimestamp": "2022-01-02T19:17:42",
        "logIndex": 318,
        "transactionHash": "0x816a39f8017fe92e80d777569055a7549b5fb38b5d1c9f34bee5d940b510ed77",
      },
      "originatorAddress": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
      "price": Object {
        "blockNumber": 13927877,
        "ethPrice": Object {
          "decimal": 0,
          "raw": "0",
        },
        "nativePrice": Object {
          "currency": Object {
            "address": "0x0000000000000000000000000000000000000000",
            "decimals": 18,
            "name": "ETH",
          },
          "decimal": 0,
        },
        "usdcPrice": Object {
          "decimal": 0,
          "raw": "0",
        },
      },
      "toAddress": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
    },
    "name": "Zorbs",
    "owner": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
    "tokenContract": Object {
      "chain": 1,
      "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
      "description": "",
      "name": "Zorbs",
      "network": "ETHEREUM",
      "symbol": "ZORB",
    },
    "tokenId": "48057",
  },
}
`;

exports[`zdk should fetch localhost collections empty object 2`] = `
Object {
  "markets": Array [
    Object {
      "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
      "marketAddress": "0x6170b3c3a54c3d8c854934cbc314ed479b2b29a3",
      "marketType": "V3_ASK",
      "networkInfo": Object {
        "chain": "MAINNET",
        "network": "ETHEREUM",
      },
      "price": Object {
        "blockNumber": 14071638,
        "ethPrice": Object {
          "decimal": 0.5,
          "raw": "500000000000000000",
        },
        "nativePrice": Object {
          "currency": Object {
            "address": "0x0000000000000000000000000000000000000000",
            "decimals": 18,
            "name": "ETH",
          },
          "decimal": 0.5,
        },
        "usdcPrice": Object {
          "decimal": 1213.445087470087,
          "raw": "1213445087470087045120",
        },
      },
      "status": "ACTIVE",
    },
  ],
  "sales": Object {},
  "token": Object {
    "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
    "content": null,
    "description": null,
    "image": null,
    "lastRefreshTime": null,
    "mintInfo": Object {
      "mintContext": Object {
        "blockNumber": 13927877,
        "blockTimestamp": "2022-01-02T19:17:42",
        "logIndex": 317,
        "transactionHash": "0x816a39f8017fe92e80d777569055a7549b5fb38b5d1c9f34bee5d940b510ed77",
      },
      "originatorAddress": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
      "price": Object {
        "blockNumber": 13927877,
        "ethPrice": Object {
          "decimal": 0,
          "raw": "0",
        },
        "nativePrice": Object {
          "currency": Object {
            "address": "0x0000000000000000000000000000000000000000",
            "decimals": 18,
            "name": "ETH",
          },
          "decimal": 0,
        },
        "usdcPrice": Object {
          "decimal": 0,
          "raw": "0",
        },
      },
      "toAddress": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
    },
    "name": "Zorbs",
    "owner": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
    "tokenContract": Object {
      "chain": 1,
      "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
      "description": "",
      "name": "Zorbs",
      "network": "ETHEREUM",
      "symbol": "ZORB",
    },
    "tokenId": "48056",
  },
}
`;

exports[`zdk should fetch localhost token full object 1`] = `
Object {
  "tokenMarkets": Object {
    "nodes": Array [
      Object {
        "markets": Array [
          Object {
            "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
            "marketAddress": "0xce6cef2a9028e1c3b21647ae3b4251038109f42a",
            "marketType": "V3_ASK",
            "networkInfo": Object {
              "chain": "MAINNET",
              "network": "ETHEREUM",
            },
            "price": Object {
              "blockNumber": 14036921,
              "ethPrice": Object {
                "decimal": 0.05,
                "raw": "50000000000000000",
              },
              "nativePrice": Object {
                "currency": Object {
                  "address": "0x0000000000000000000000000000000000000000",
                  "decimals": 18,
                  "name": "ETH",
                },
                "decimal": 0.05,
              },
              "usdcPrice": Object {
                "decimal": 156.40084904014552,
                "raw": "156400849040145514496",
              },
            },
            "properties": Object {
              "__typename": "V3Ask",
              "askPrice": Object {
                "blockNumber": 14036921,
                "ethPrice": Object {
                  "decimal": 0.05,
                  "raw": "50000000000000000",
                },
                "nativePrice": Object {
                  "currency": Object {
                    "address": "0x0000000000000000000000000000000000000000",
                    "decimals": 18,
                    "name": "ETH",
                  },
                  "decimal": 0.05,
                },
                "usdcPrice": Object {
                  "decimal": 156.40084904014552,
                  "raw": "156400849040145514496",
                },
              },
              "askStatus": "ACTIVE",
              "buyer": null,
              "finder": null,
              "findersFeeBps": 100,
              "seller": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
              "sellerFundsRecipient": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
            },
            "status": "ACTIVE",
          },
        ],
        "sales": Object {},
        "token": Object {
          "attributes": null,
          "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
          "content": null,
          "description": null,
          "image": null,
          "lastRefreshTime": null,
          "metadata": null,
          "mintInfo": Object {
            "mintContext": Object {
              "blockNumber": 13927877,
              "blockTimestamp": "2022-01-02T19:17:42",
              "logIndex": 318,
              "transactionHash": "0x816a39f8017fe92e80d777569055a7549b5fb38b5d1c9f34bee5d940b510ed77",
            },
            "originatorAddress": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
            "price": Object {
              "blockNumber": 13927877,
              "ethPrice": Object {
                "decimal": 0,
                "raw": "0",
              },
              "nativePrice": Object {
                "currency": Object {
                  "address": "0x0000000000000000000000000000000000000000",
                  "decimals": 18,
                  "name": "ETH",
                },
                "decimal": 0,
              },
              "usdcPrice": Object {
                "decimal": 0,
                "raw": "0",
              },
            },
            "toAddress": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
          },
          "name": "Zorbs",
          "owner": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
          "tokenContract": Object {
            "chain": 1,
            "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
            "description": "",
            "name": "Zorbs",
            "network": "ETHEREUM",
            "symbol": "ZORB",
          },
          "tokenId": "48057",
          "tokenUrl": "data:application/json;base64,eyJuYW1lIjogIlpvcmIgIzQ4MDU3IiwgImRlc2NyaXB0aW9uIjogIlpvcmJzIHdlcmUgZGlzdHJpYnV0ZWQgZm9yIGZyZWUgYnkgWk9SQSBvbiBOZXcgWWVhcuKAmXMgMjAyMi4gRWFjaCBORlQgaW1idWVzIHRoZSBwcm9wZXJ0aWVzIG9mIGl0cyB3YWxsZXQgaG9sZGVyLCBhbmQgd2hlbiBzZW50IHRvIHNvbWVvbmUgZWxzZSwgd2lsbCB0cmFuc2Zvcm0uXG5cblZpZXcgdGhpcyBORlQgYXQgW3pvcmIuZGV2L25mdC80ODA1N10oaHR0cHM6Ly96b3JiLmRldi9uZnQvNDgwNTcpIiwgImltYWdlIjogImRhdGE6aW1hZ2Uvc3ZnK3htbDtiYXNlNjQsUEhOMlp5QjRiV3h1Y3owaWFIUjBjRG92TDNkM2R5NTNNeTV2Y21jdk1qQXdNQzl6ZG1jaUlIWnBaWGRDYjNnOUlqQWdNQ0F4TVRBZ01URXdJajQ4WkdWbWN6NDhjbUZrYVdGc1IzSmhaR2xsYm5RZ2FXUTlJbWQ2Y2lJZ1ozSmhaR2xsYm5SVWNtRnVjMlp2Y20wOUluUnlZVzV6YkdGMFpTZzJOaTQwTlRjNElESTBMak0xTnpVcElITmpZV3hsS0RjMUxqSTVNRGdwSWlCbmNtRmthV1Z1ZEZWdWFYUnpQU0oxYzJWeVUzQmhZMlZQYmxWelpTSWdjajBpTVNJZ1kzZzlJakFpSUdONVBTSXdKU0krUEhOMGIzQWdiMlptYzJWMFBTSXhOUzQyTWlVaUlITjBiM0F0WTI5c2IzSTlJbWh6YkNneE5qTXNJRGN4SlN3Z09EY2xLU0lnTHo0OGMzUnZjQ0J2Wm1aelpYUTlJak01TGpVNEpTSWdjM1J2Y0MxamIyeHZjajBpYUhOc0tERTJNeXdnTnpNbExDQTNNU1VwSWlBdlBqeHpkRzl3SUc5bVpuTmxkRDBpTnpJdU9USWxJaUJ6ZEc5d0xXTnZiRzl5UFNKb2Myd29NVGt3TENBM055VXNJRFE1SlNraUlDOCtQSE4wYjNBZ2IyWm1jMlYwUFNJNU1DNDJNeVVpSUhOMGIzQXRZMjlzYjNJOUltaHpiQ2d4T1RJc0lEZzNKU3dnTkRFbEtTSWdMejQ4YzNSdmNDQnZabVp6WlhROUlqRXdNQ1VpSUhOMGIzQXRZMjlzYjNJOUltaHpiQ2d4T1RJc0lEa3hKU3dnTkRBbEtTSWdMejQ4TDNKaFpHbGhiRWR5WVdScFpXNTBQand2WkdWbWN6NDhaeUIwY21GdWMyWnZjbTA5SW5SeVlXNXpiR0YwWlNnMUxEVXBJajQ4Y0dGMGFDQmtQU0pOTVRBd0lEVXdRekV3TUNBeU1pNHpPRFU0SURjM0xqWXhORElnTUNBMU1DQXdRekl5TGpNNE5UZ2dNQ0F3SURJeUxqTTROVGdnTUNBMU1FTXdJRGMzTGpZeE5ESWdNakl1TXpnMU9DQXhNREFnTlRBZ01UQXdRemMzTGpZeE5ESWdNVEF3SURFd01DQTNOeTQyTVRReUlERXdNQ0ExTUZvaUlHWnBiR3c5SW5WeWJDZ2paM3B5S1NJZ0x6NDhjR0YwYUNCemRISnZhMlU5SW5KblltRW9NQ3d3TERBc01DNHdOelVwSWlCbWFXeHNQU0owY21GdWMzQmhjbVZ1ZENJZ2MzUnliMnRsTFhkcFpIUm9QU0l4SWlCa1BTSk5OVEFzTUM0MVl6STNMak1zTUN3ME9TNDFMREl5TGpJc05Ea3VOU3cwT1M0MVV6YzNMak1zT1RrdU5TdzFNQ3c1T1M0MVV6QXVOU3czTnk0ekxEQXVOU3cxTUZNeU1pNDNMREF1TlN3MU1Dd3dMalY2SWlBdlBqd3ZaejQ4TDNOMlp6ND0ifQ==",
          "tokenUrlMimeType": null,
        },
      },
    ],
    "pageInfo": Object {
      "limit": 200,
      "offset": 0,
    },
  },
}
`;
